docs/source/dev/py_postprocessing.rst

```diff
@@ -0,0 +1,178 @@
+
+Each reader class needs to implement the following interface functions:
+
+.. code:: python
+
+   def __init__(self, run_directory)
+
+   def get_data_path(self, your_specific_arguments)
+       # has to return the path to the underlying data file
+       # specified by some specific arguments (depending on the plugin output)
+
+   def get_iterations(self, your_specific_arguments)
+       # return a numpy array with iteration numbers for which data is available
+
+   def get(self, your_specific_arguments)
+       # main function of the reader to actually get the data
+
+Visualizer
+~~~~~~~~~~
+The visualizers should reside in the ``lib/python/picongpu/plugins/plot_mpl/`` directory.
+The module names should end on ``_visualizer.py`` and the class name should only be ``Visualizer``.
+
+There is a base class for visualization found in ``base_visualizer.py`` which already handles the plotting logic. It uses the
+data reader classes for accessing the data.
+After getting the data, it ensures that (for performance reasons) a matplotlib artist is created only for the first plot and later only gets updated with fresh data.
+
+The complete implementation logic is pretty simple.
+
+.. code:: python
+
+   def visualize(self, ax=None, **kwargs):
+        """
+        1. Creates the 'plt_obj' if it does not exist
+        2. Fills the 'data' parameter by using the reader
+        3. Updates the 'plt_obj' with the new data.
+        """
+        if ax is None:
+            raise ValueError("A matplotlib axes object needs to be passed!")
+
+        self.data = self.data_reader.get(**kwargs)
+        if self.plt_obj is None:
+            self._create_plt_obj(ax)
+        else:
+            self._update_plt_obj()
+
+
+All new plugins should derive from this class.
+
+When implementing a new visualizer you have to perform the following steps:
+
+1. Let your visualizer class inherit from the ``Visualizer`` class in ``base visualizer.py``.
+
+2. Implement the ``_create_data_reader(self, run_directory)`` function. This function should return a data reader object (see above)
+   for this plugins data.
+
+3. Implement the ``_create_plt_obj(self, ax)`` function. This function needs to access the plotting data from the ``self.data`` member (this is the data structure as returned
+by the data readers ``.get(...)`` function, create some kind of matplotlib artist by storing it in the ``self.plt_obj`` member variable and set up other plotting details
+(e.g. a colorbar).
+
+4. Implement the ``_update_plt_obj(self)`` function. This is called only after a valid ``self.plt_obj`` was created. It updates the matplotlib artist with new data.
+Therefore it again needs to access the plotting data from the ``self.data`` member and call the data update API for the matplotlib artist (normally via ``.set_data(...)``.
+
+Jupyter widget
+~~~~~~~~~~~~~~
+The widget is essentially only a wrapper around the matplotlib visualizer that allows dynamical adjustment
```

Comment:
please keep prose in one line per sentence.
This sentence is also too long. As a suggestion, put the example just as a new sentence :)